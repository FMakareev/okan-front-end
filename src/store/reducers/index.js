import { combineReducers } from 'redux';
import { reducer as formReducer } from 'redux-form';
// import { localizeReducer } from 'react-localize-redux';

export const rootReducer = combineReducers({
  form: formReducer,
  // locale: localizeReducer,
});

export default rootReducer;
