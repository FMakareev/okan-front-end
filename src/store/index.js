/* global isBrowser */
import { createStore, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';

import { composeWithDevTools } from 'redux-devtools-extension';
import { rootReducer } from './reducers/index';

const createStoreMiddleware = applyMiddleware(thunkMiddleware)(createStore);
const initialState = isBrowser ? window.PRELOADED_REDUX_STATE : {};

const Store = createStoreMiddleware(rootReducer, initialState, composeWithDevTools());

export { Store };

export default Store;
