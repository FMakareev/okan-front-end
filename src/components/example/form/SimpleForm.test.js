import React, {Component} from 'react';
import renderer from 'react-test-renderer';
import ReactTestUtils from 'react-dom/test-utils'

import { Provider as ProviderRedux } from 'react-redux';

import SimpleForm, {RenderField} from './SimpleForm';
import {Store} from "../../../store";


test('SimpleForm: просто рендер', () => {
  const output = renderer.create(
    <ProviderRedux store={Store}>
      <SimpleForm/>
    </ProviderRedux>
  ).toJSON();
  expect(output).toMatchSnapshot();
});

test('SimpleForm: onSubmit должен вызватся 1 раз', () => {
  const spy = jest.fn();

  const wrapper  =  renderer.create(
    <ProviderRedux store={Store}>
      <SimpleForm
        handleSubmit={spy}
      />
    </ProviderRedux>
  );
  const tree = wrapper.toJSON();
  tree.props.onSubmit();

  expect(spy).toHaveBeenCalledTimes(1);
});
test('SimpleForm: проверка инифиализации формы с данными', () => {
  const initialValues = {
    firstName: 'First Name',
    lastName: 'Last Name',
    email: 'email@email.com',
    sex: 'male',
  };
  const wrapper = renderer.create(
    <ProviderRedux store={Store}>
      <SimpleForm
        initialValues={initialValues}
      />
    </ProviderRedux>
  );

  // сравниваем то что находится в redux с тем что мы передали
  expect(Store.getState().form.simple.values).toEqual(initialValues);
});
test('SimpleForm: проверка инифиализации формы с данными', () => {
  const initialValues = {
    firstName: 'First Name',
    lastName: 'Last Name',
    email: 'email@email.com',
    sex: 'male',
  };
  const wrapper = renderer.create(
    <ProviderRedux store={Store}>
      <SimpleForm
        initialValues={initialValues}
      />
    </ProviderRedux>
  );

  // сравниваем то что находится в redux с тем что мы передали
  expect(Store.getState().form.simple.values).toEqual(initialValues);
});

test('SimpleForm: вызов onChange на компоненте и name = firstName', () => {
  const initialValues = {
    firstName: 'First Name',
    lastName: 'Last Name',
    email: 'email@email.com',
    sex: 'male',
  };
  const dom = ReactTestUtils.renderIntoDocument(
    <ProviderRedux store={Store}>
      <SimpleForm
        initialValues={initialValues}
      />
    </ProviderRedux>
  );
  // https://reactjs.org/docs/test-utils.html#findrenderedcomponentwithtype
  let props = ReactTestUtils.findRenderedComponentWithType(dom, RenderField).props;
  props.input.onChange('Bla bla bla');
  props = ReactTestUtils.findRenderedComponentWithType(dom, RenderField).props;
  expect(props.input.value).toBe('Bla bla bla');
});


// https://github.com/erikras/redux-form/blob/master/src/__tests__/Form.spec.js
