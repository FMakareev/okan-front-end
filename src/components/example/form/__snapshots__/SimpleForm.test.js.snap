// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleForm: просто рендер 1`] = `
<form
  onSubmit={[Function]}
>
  <div>
    <label>
      First Name
    </label>
    <div>
      <input
        name="firstName"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder="First Name"
        type="text"
        value=""
      />
    </div>
  </div>
  <div>
    <label>
      Last Name
    </label>
    <div>
      <input
        name="lastName"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder="Last Name"
        type="text"
        value=""
      />
    </div>
  </div>
  <div>
    <label>
      Email
    </label>
    <div>
      <input
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder="Email"
        type="email"
        value=""
      />
    </div>
  </div>
  <div>
    <label>
      Sex
    </label>
    <div>
      <label>
        <input
          checked={false}
          name="sex"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          type="radio"
          value="male"
        />
         
        Male
      </label>
      <label>
        <input
          checked={false}
          name="sex"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          type="radio"
          value="female"
        />
         
        Female
      </label>
    </div>
  </div>
  <div>
    <label>
      Favorite Color
    </label>
    <div>
      <select
        name="favoriteColor"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        value=""
      >
        <option />
        <option
          value="ff0000"
        >
          Red
        </option>
        <option
          value="00ff00"
        >
          Green
        </option>
        <option
          value="0000ff"
        >
          Blue
        </option>
      </select>
    </div>
  </div>
  <div>
    <label
      htmlFor="employed"
    >
      Employed
    </label>
    <div>
      <input
        checked={false}
        id="employed"
        name="employed"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        type="checkbox"
        value=""
      />
    </div>
  </div>
  <div>
    <label>
      Notes
    </label>
    <div>
      <textarea
        name="notes"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        value=""
      />
    </div>
  </div>
  <div>
    <button
      disabled={true}
      type="submit"
    >
      Submit
    </button>
    <button
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      Clear Values
    </button>
  </div>
</form>
`;
